cocos/2d/CCFontFNT.cpp

```diff
@@ -809,7 +809,11 @@ FontAtlas * FontFNT::createFontAtlas()
         tempDefinition.validDefinition = true;
         tempDefinition.xAdvance = fontDef.xAdvance;
         // add the new definition
-        tempAtlas->addLetterDefinition(fontDef.charID,tempDefinition);
+        if (65535 < fontDef.charID) {
```

Comment:
I'm not sure.  The problem is that if .fnt file includes 'id' equals or greater than 65536, the upper bits of the value will be lost because of down casting to char16_t, the first arg of FontAtlas::addLetterDefinition method function. The value is used as a key for _letterDefinitions map.  Therefore, if 'id' equals to 65633, the value will override an entry of _letterDefinitions for character 'a'.  This made me trouble to find the problem in my .fnt file.  Just produce warning message will save our time.